src/du/native/clonesize.c

```
@@ -760,42 +1515,300 @@ static void print_human(const char *target, const Result *R, int quiet) {
 // bun:ffi entry point — one-shot scan, returns a malloc'd JSON string.
 // Caller must clonesize_free() it. Not reentrant (uses process globals).
 // ---------------------------------------------------------------------------
+// Reset every accumulating global — the dylib lives across calls under bun:ffi,
+// so anything left over from the previous scan would be double-counted.
+static void reset_state(void) {
+    for (int i = 0; i < g_ngroups; i++) { free(g_group_names[i]); g_group_names[i] = NULL; }
+    g_ngroups = 0;
+    memset(g_group_naive, 0, sizeof g_group_naive);
+    memset(g_group_files, 0, sizeof g_group_files);
+    memset(g_group_private, 0, sizeof g_group_private);
+    for (int i = 0; i < g_nnodes; i++) free(g_nodes[i].name);
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)